virtcontainers/clh.go

```diff
@@ -410,6 +412,38 @@ func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
 	return vcpuInfo, nil
 }
 
+func (clh *cloudHypervisor) hotplugBlockDevice(drive *config.BlockDrive) error {
+	if clh.config.BlockDeviceDriver != config.VirtioBlock {
+		return fmt.Errorf("incorrect hypervisor configuration on 'block_device_driver':"+
+			" using '%v' but only support 'virtio-blk'", clh.config.BlockDeviceDriver)
+	}
+
+	cl := clh.client()
+	ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+	defer cancel()
+
+	_, _, err := cl.VmmPingGet(ctx)
+	if err != nil {
+		return openAPIClientError(err)
+	}
+
+	if drive.Pmem {
+		err = fmt.Errorf("pmem device hotplug not supported")
```

Comment:
qq. is nvdimm hotplug supported?

Reply:
It is, I think @likebreath  is splitting in two PRs

Reply:
@devimc Right. I am planning to have another PR for pmem hotplug. The changes to clh driver should be simple, but I will need to understand how I can validate the pmem hotplug capability. @jcvenegas told me that @GabyCT is working on a pmem hotplug test which might be leveraged. I will get back to it soon. :)